Polygon_mesh_processing/doc/Polygon_mesh_processing/Polygon_mesh_processing.txt

```diff
@@ -511,6 +511,37 @@ input triangle mesh.
 \subsubsection InsideExample Inside Test Example
 \cgalExample{Polygon_mesh_processing/point_inside_example.cpp}
 
+
+\subsection PMPEnvelope Polyhedral Envelope Containment Check
+
+The class `CGAL::Envelope` provides functors that test whether a query point, segment or triangle
+is fully contained in a <em>polyhedral envelope</em> of a triangle mesh.
+
+\cgalFigureBegin{envelopeFig, envelope.png}
+Two polyhedral envelopes.
+\cgalFigureEnd
+
+While the user provides a distance
+the envelope is not the Minkowski sum with a sphere, because this would have cylindrical and spherical
+patches at convex edges and vertices of the input triangle mesh. Instead, the envelope consists
+of a collection of prisms that are guaranteed to be inside the Minkowski sum envelope, hence the
+term polyhedral envelope.  This containment test is exact for the polyhedral envelope, and
+conservative in the sense that, if a query is inside the polyhedral envelope, we can be sure
+that it is also in the Minkowski sum envelope, but if it is outside the polyhedral envelop
+we do not know where it is with respect to the Minkowski sum envelope.
+
+Such a test is used by the class `Surface_mesh_simplification::Envelope_filter` of the
+package \refPkgSurfaceMeshSimplification, in order to simplify respecting a tolerance.
+
+\subsubsection InsideExample Polyhedral Envelope Example
+\cgalExample{Polygon_mesh_processing/polyhedral_envelope.cpp}
+
+The algorithm is described in \cgalCite{cgal:Wwshap-eepec-20} and can be summarized as follows.
+First for each triangle a <em>prism</em> is constructed with an offset of \f$ \epsilon/\sqrt(3) \f$ where \f$ \epsilon\f$ is the tolerance,
+and all prisms are stored in an AABB tree.  For a query point the algorithm checks if is inside a prism.
+For a query segment or triangle the algorithms checks if the query is completely covered.
+The details of how to check this covering can be found in the paper of Wang et al.
```

Comment:
```suggestion
The details of how to check this covering can be found in \cgalCite{cgal:Wwshap-eepec-20}.
```

Reply:
We have that \cite already in the first sentence of the paragraph.

Reply:
I moved the \cite to the first sentence.